include/heuristic/HeuristicMapper.hpp

```diff
@@ -151,25 +150,44 @@ class HeuristicMapper : public Mapper {
       swaps.back().emplace_back(source, target, middle_anc, qc::Teleportation);
     }
 
-    void updateHeuristicCost(const Architecture& arch, const Gate& gate,
-                             bool admissibleHeuristic) {
-      auto cost =
-          arch.distance(locations.at(gate.control), locations.at(gate.target));
-      if (admissibleHeuristic) {
-        costHeur = std::max(costHeur, cost);
-      } else {
-        costHeur += cost;
-      }
-    }
-
     /**
-     * @brief checks if the qubits of the given gate are mapped next to each
-     * other. If not, set `done` to false
+     * @brief calculates the heuristic cost of the current mapping in the node
+     * for some given layer and writes it to `Node::costHeur` additional
+     * `Node::done` is set to true if all qubits shared by a gate in the layer
+     * are mapped next to each other
+     *
+     * @param arch the architecture for calculating distances between physical
+     * qubits
+     * @param currentLayer a vector of all gates in the current layer
+     * @param admissibleHeuristic controls if the heuristic should be calculated
+     * such that it is admissible (i.e. A*-search should yield the optimal
+     * solution using this heuristic)
+     * @param considerFidelity controls if the heuristic should consider
+     * fidelity data of the architecture
      */
-    void checkUnfinished(const Architecture& arch, const Gate& gate) {
-      if (arch.distance(locations.at(gate.control), locations.at(gate.target)) >
-          COST_DIRECTION_REVERSE) {
-        done = false;
+    void updateHeuristicCost(const Architecture&      arch,
+                             const std::vector<Gate>& currentLayer,
+                             bool admissibleHeuristic, bool considerFidelity) {
+      costHeur = 0.;
+      done     = true;
+      for (const auto& gate : currentLayer) {
+        if (gate.singleQubit())
+          continue;
+
+        auto cost          = arch.distance(locations.at(gate.control),
+                                           locations.at(gate.target));
+        auto fidelity_cost = cost;
+        if (considerFidelity) {
+        }
```

Comment:
These two lines can safely be removed in this PR as they do not contribute any functionality. (which is also why GitHub CodeQL is complaining)
```suggestion
```